test/module/irohad/multi_sig_transactions/gossip_propagation_strategy_test.cpp

```diff
@@ -91,16 +117,13 @@ TEST(GossipPropagationStrategyTest, EmittingAllPeers) {
   std::vector<std::string> peersId;
   PropagationData peers = generate(peersId, peers_size);
 
-  auto emitted = subscribe_and_emit(peers, 1ms, amount, take);
+  auto emitted = subscribeAndEmit(peers, 1ms, amount, take);
 
   // emitted.size() can be less than peers.size()
   ASSERT_GE(peers.size(), emitted.size());
   // because emitted size should be increased by amount at once
   ASSERT_FALSE(emitted.size() % amount);
-  std::for_each(emitted.begin(), emitted.end(), [&peersId](const auto &v) {
-    ASSERT_NE(std::find(peersId.begin(), peersId.end(), v.address),
-              peersId.end());
-  });
+  validateEmitted(emitted, peersId);
```

Comment:
This approach will show line 104 in gtest output, and it will not be possible to easily go to line where assertion has failed. Returning `bool` from `validateEmitted`, and writing an assertion here would be more informative for end user.